src/fu-engine.c

```diff
@@ -5944,6 +5946,16 @@ fu_engine_ensure_security_attrs(FuEngine *self)
 	/* distil into one simple string */
 	g_free(self->host_security_id);
 	self->host_security_id = fu_engine_attrs_calculate_hsi_for_chassis(self);
+
+	/* Convert Security attribute to json string */
+	data = fu_security_attrs_to_json_string(self->host_security_attrs, &error);
+
+	/* Store string to db */
+	if (data != NULL)
+		fu_history_add_security_attribute(self->history,
```

Comment:
you should check the return of this function and display the error if it failed as a warning

Reply:
Sure, An DB writing fail message will be added here.